gdb/gdb.go

```diff
@@ -1,198 +1,285 @@
+		if apiGenre, ok := allGenres[strconv.Itoa(int(genreID))]; ok {
+			genres = append(genres, ParsedGenre{
+				ID:   int(apiGenre.Id),
+				Name: apiGenre.Name,
+			})
+		}
 	}
-	u.RawQuery = q.Encode()
-	hReq, err := http.NewRequest("GET", u.String(), nil)
-	if err != nil {
-		return nil, err
+	res.Genres = genres
+
+	allDevelopers := getCachedDevelopers(ctx, apikey)
+	developers := []ParsedDeveloper{}
+	for _, developerID := range apiGame.Developers {
+		if apiDeveloper, ok := allDevelopers[strconv.Itoa(int(developerID))]; ok {
+			developers = append(developers, ParsedDeveloper{
+				ID:   int(apiDeveloper.Id),
+				Name: apiDeveloper.Name,
+			})
+		}
 	}
-	hReq = hReq.WithContext(ctx)
-	resp, err := http.DefaultClient.Do(hReq)
-	if err != nil {
-		return nil, fmt.Errorf("getting game list url:%s, error:%s", u, err)
+	res.Developers = developers
+
+	allPublishers := getCachedPublishers(ctx, apikey)
+	publishers := []ParsedPublisher{}
+	for _, publisherID := range apiGame.Publishers {
+		if apiPublisher, ok := allPublishers[strconv.Itoa(int(publisherID))]; ok {
+			publishers = append(publishers, ParsedPublisher{
+				ID:   int(apiPublisher.Id),
+				Name: apiPublisher.Name,
+			})
+		}
 	}
-	defer resp.Body.Close()
-	r := &GGLResp{}
-	decoder := xml.NewDecoder(resp.Body)
-	if err := decoder.Decode(r); err != nil {
-		return nil, err
+	res.Publishers = publishers
+
+	for range games.Data.Games {
+		getCachedGenres(ctx, apikey)
 	}
-	if r.XMLName.Local == "Error" {
-		return nil, fmt.Errorf("GetGameList error: %s", r.Err)
+
+	images, _, err := apiClient.GamesApi.GamesImages(ctx, apikey, strconv.Itoa(res.ID), nil)
+	if err == nil {
+		res.ImageBaseUrls = ParsedImageSizeBaseUrls{
+			Original: images.Data.BaseUrl.Original,
+			Thumb:    images.Data.BaseUrl.Thumb,
+		}
+
+		parsedImages := make(map[string][]ParsedGameImage)
+		for key, val := range images.Data.Images {
+			result := parsedImages[key]
+			for _, image := range val {
+				result = append(result, ParsedGameImage{
+					ID:       int(image.Id),
+					Type:     image.Type,
+					Side:     image.Side,
+					Filename: image.Filename,
+				})
+			}
+			parsedImages[key] = result
+		}
+		res.Images = parsedImages
 	}
-	r.Err = ""
-	return r, nil
+
+	return res, nil
 }
 
 // IsUp returns if thegamedb.net is up.
-func IsUp(ctx context.Context) bool {
-	u, err := url.Parse(gdbURL)
-	u.Path = ggPath
-	q := url.Values{}
-	q.Set("id", "1")
-	u.RawQuery = q.Encode()
-	hReq, err := http.NewRequest("GET", u.String(), nil)
-	if err != nil {
-		return false
-	}
-	hReq = hReq.WithContext(ctx)
-	resp, err := http.DefaultClient.Do(hReq)
-	if err != nil {
-		return false
-	}
-	if resp.StatusCode != 200 {
-		return false
-	}
-	defer resp.Body.Close()
-	r := &GGResp{}
-	decoder := xml.NewDecoder(resp.Body)
-	if err := decoder.Decode(r); err != nil {
+func IsUp(ctx context.Context, apikey string) bool {
+	_, resp, err := apiClient.GamesApi.GamesByGameID(ctx, apikey, "1", nil)
```

Comment:
Why are all the ints represented as strings from the new API?

Reply:
For the responses, I'm not sure. Could ask on their discord, @Zer0xFF was really helpful on other questions I had there.

For the requests, its technically a csv of ids so string makes sense. Might be able to finagle the swagger spec to encode that it is an array of ints but I'm pretty new to it so I'm not sure.